pkg/scope/provider.go

```diff
@@ -63,10 +68,14 @@ func (providerScopeFactory) NewClientScopeFromMachine(ctx context.Context, ctrlC
 		caCert = defaultCACert
 	}
 
-	return NewProviderScope(cloud, caCert, logger)
+	if f.cache == nil {
```

Comment:
Should we check for `== 0` here, as it says "Setting this value to 0 means no cache."?